plugin/trino-hive/src/main/java/io/trino/plugin/hive/aws/athena/PartitionProjectionService.java

```diff
@@ -363,7 +357,17 @@ private <I, V> void rewriteProperty(
             String targetPropertyKey,
             Function<I, V> valueMapper)
     {
-        Optional.ofNullable(sourceProperties.get(sourcePropertyKey))
+        rewriteProperty(sourceProperties::get, targetPropertiesBuilder, sourcePropertyKey, targetPropertyKey, valueMapper);
+    }
+
+    private <I, V> void rewriteProperty(
+            Function<String, I> sourcePropertyProvider,
+            ImmutableMap.Builder<String, V> targetPropertiesBuilder,
+            String sourcePropertyKey,
+            String targetPropertyKey,
+            Function<I, V> valueMapper)
+    {
+        Optional.ofNullable(sourcePropertyProvider.apply(sourcePropertyKey))
```

Comment:
nit: I believe making optional to simply check if something is not null should be replaced with a simple if for readability. But this has not been changed here, so it's ok for me to leave it